frontend/src/hocs/withMapboxToken/MapboxToken.ts

```diff
@@ -34,11 +41,29 @@ export class MapboxToken {
    * only be fetched once per session.)
    */
   public static async get(): Promise<string> {
-    if (MapboxToken.token == null) {
+    if (!MapboxToken.token) {
       if (SessionInfo.current.userMapboxToken !== "") {
         MapboxToken.token = SessionInfo.current.userMapboxToken
       } else {
-        MapboxToken.token = await this.fetchToken(TOKENS_URL, "mapbox")
+        const { commandLine } = SessionInfo.current
+
+        if (
+          this.isItRunningLocal() &&
+          commandLine.toLowerCase() === "streamlit hello"
+        ) {
+          MapboxToken.token = await this.fetchToken(TOKENS_URL, "mapbox")
+        } else {
+          throw new Error(
+            `
+            To use this you'll need a Mapbox access token. Please add it to your config.
+            
+            To get a token for yourself, create an account at
+            <a href="https://mapbox.com">https://mapbox.com</a>. It's free! (for moderate usage levels) See
+            <a href="https://docs.streamlit.io/cli.html#view-all-config-options">our documentation</a> for more
+            info on how to set config options.
+            `
```

Comment:
Hey, I was going to comment on this error message, but I need to see it in action to really understand what it should say. Can you give me an example of how to see it?

Reply:
you could run for example `/e2e/scripts/st_map.py` without setting any mapbox token within your config.toml

Reply:
This is how it looks
<img width="750" alt="Screen Shot 2020-04-30 at 2 19 27 PM" src="https://user-images.githubusercontent.com/631997/80739921-ad717e80-8aed-11ea-97bb-dd397cfbaed0.png">

Reply:
I see the following snippet when I run `streamlit config view`:
```
[mapbox]

# Configure Streamlit to use a custom Mapbox token for elements like st.deck_gl_chart and st.map. If you don't do this you'll be using Streamlit's own token, which has limitations and is not guaranteed to always work. To get a token for yourself, create an account at https://mapbox.com. It's free! (for moderate usage levels)
# Default: ""
token = ""
```

I wonder if we need to change the comment in the config, because with these change the user will no longer be using Streamlit's own token unless they are running `streamlit hello` locally.

EDIT: I believe the config comments are at https://github.com/streamlit/streamlit/blob/908827b8230222621e385a3ee548d529ed9e740d/lib/streamlit/config.py#L538

Reply:
Good catch!, thank you @Amey-D